File: survey.js
function analyzeSurvey(data) {
  // Group engagement scores by department
  const departmentScores = {};
  data.forEach((entry) => {
    if (!departmentScores[entry.department]) {
      departmentScores[entry.department] = [];
    }
    departmentScores[entry.department].push(entry.engagement_score);
  });

  // Calculate average engagement scores for each department
  const engagementScores = {};
  for (const department in departmentScores) {
    const scores = departmentScores[department];
    const average = scores.reduce((a, b) => a + b, 0) / scores.length;
    engagementScores[department] = average;
  }

  // Identify department with highest and lowest engagement
  let highestEngagement = null,
    lowestEngagement = null;
  for (const department in engagementScores) {
    if (
      highestEngagement === null ||
      engagementScores[department] > engagementScores[highestEngagement]
    ) {
      highestEngagement = department;
    }
    if (
      lowestEngagement === null ||
      engagementScores[department] < engagementScores[lowestEngagement]
    ) {
      lowestEngagement = department;
    }
  }

  return {
    engagementScores,
    highestEngagement,
    lowestEngagement,
  };
}

// Example usage
const surveyData = [
  { employee_id: 1, department: "HR", engagement_score: 8 },
  { employee_id: 2, department: "Engineering", engagement_score: 7 },
  { employee_id: 3, department: "HR", engagement_score: 6 },
  { employee_id: 4, department: "Engineering", engagement_score: 9 },
];
console.log(analyzeSurvey(surveyData));

File: old/survey.py
import pandas as pd

def analyze_survey(data):
    # Load the data
    df = pd.DataFrame(data)
    
    # Basic cleaning: Drop missing values
    df = df.dropna()
    
    # Aggregate engagement scores by department
    engagement_scores = df.groupby('department')['engagement_score'].mean().to_dict()

    print(engagement_scores)

    # Identify department with highest and lowest engagement
    highest_engagement = max(engagement_scores, key=engagement_scores.get)
    lowest_engagement = min(engagement_scores, key=engagement_scores.get)
    
    return {
        "engagement_scores": engagement_scores,
        "highest_engagement": highest_engagement,
        "lowest_engagement": lowest_engagement
    }

# Example usage
survey_data = [
    {"employee_id": 1, "department": "HR", "engagement_score": 8},
    {"employee_id": 2, "department": "Engineering", "engagement_score": 7},
    {"employee_id": 3, "department": "HR", "engagement_score": 6},
    {"employee_id": 4, "department": "Engineering", "engagement_score": 9}
]
print(analyze_survey(survey_data))
